src/content_sao.cpp

```diff
@@ -402,15 +402,27 @@ void LuaEntitySAO::step(float dtime, bool send_recommended)
 		}
 	}
 
-	if(m_registered){
+	if (m_registered) {
 		m_env->getScriptIface()->luaentity_Step(m_id, dtime);
 	}
 
-	if(send_recommended == false)
+	// Remove LuaEntity outside of mapgen limits
+	{
+		ServerMap *map = dynamic_cast<ServerMap *>(&m_env->getMap());
+		assert(map);
+		if (!m_pending_deactivation && map->sao_pos_over_mapgen_limit(m_base_position)) {
+			errorstream << "Remove SAO " << m_id << "(" << m_init_name
```

Comment:
yeah i will change this before merge, it was for testing purpose, i think infostream is sufficient here